### DISCUSSION TITLE
Webpack alias not working

### DISCUSSION THREAD

**Discussion by robertontiu** [vercel/next.js#15624] [Help] [answered]
# Bug report

## Describe the bug

I am working on a project that consists of three apps - one react native (w/ expo), one nextjs and one storybook for web. All three share a couple of libraries (project is a monorepo) that are built for react-native but use react-native-web in nextjs and storybook (via webpack alias). In order to eliminate issues with expo modules on web I have added webpack aliases that would resolve to some mocked implementations we've created. While everything works perfectly on the storybook app, nextjs still resolve to the original module instead of the aliases and I have no idea what to do.

PS. Project is typescript but adding expo mocks to tsconfig paths does not fix the issue.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

next.config.js
```
const path = require('path')
const withPlugins = require('next-compose-plugins')
const transpileModules = require('next-transpile-modules')
const withModules = transpileModules(['@falcon/components', '@falcon/theming', '@falcon/config', '@falcon/icons'])
const {setupWebpack} = require('@falcon/mock')

module.exports = withPlugins([withModules], {
	webpack: config => {
		config.resolve.extensions = ['.web.js', '.web.ts', '.web.tsx', ...config.resolve.extensions]
		return setupWebpack(config)
	},
	sassOptions: {
		includePaths: [path.join(__dirname, 'styles')],
	},
})
```

@falcon/mock/setupWebpack.js
```
const alias = {
    'react-native$': 'react-native-web',
    'expo': '@falcon/mock/adapters/noop',
    'expo-font': '@falcon/mock/adapters/noop',
    '@expo-google-fonts/montserrat': '@falcon/mock/adapters/noop',
    'expo-constants': '@falcon/mock/adapters/expo-constants'
}

module.exports = function setupWebpack(config) {
    config.resolve.alias = {
        ...config.resolve.alias,
        ...Object.keys(alias).reduce((acc, key) => {
            acc[key] =  require.resolve(alias[key])
            return acc
        }, {})
    }

    return config
}
```

## Expected behavior

Aliases should resolve as defined.

## Screenshots

`expo` still resolves to the original path
![image](https://user-images.githubusercontent.com/9152337/88783557-3a7cb000-d18f-11ea-8604-8671a8f48ff6.png)

## System information

- OS: Windows
- Version of Next.js: 9.4.4
- Version of Node.js: 10.21.0

**Comment by timneutkens** [comment, marked as answer]
Next.js does not compile node_modules by default so you can't add in aliases to webpack that have to be resolved inside of npm packages. I'll convert this to a discussion given that it's not a bug.

You'll probably have to add `expo` to `transpileModules`

**Reply by robertontiu** [reply]
You are right - that fixes the issue. Thank you!

**Reply by restcristian** [reply]
@timneutkens So that means that if I have a dependency, let's say Redux Toolkit, and that inside uses React-Redux, I cannot add an alias to make sure it uses another packages instead of react-redux?
